crates/bestool/src/actions/crypto/decrypt.rs

```diff
@@ -22,16 +24,32 @@ pub async fn run(ctx: Context<CryptoArgs, DecryptArgs>) -> Result<()> {
 		.parse()
 		.map_err(|err: &str| miette!("failed to parse: {err}"))?;
 
-	let encrypted = tokio::fs::read(&ctx.args_sub.encrypted)
+	let encrypted = tokio::fs::File::open(&ctx.args_sub.encrypted)
 		.await
 		.into_diagnostic()
-		.wrap_err("reading the encrypted file")?;
+		.wrap_err("opening the encrypted file")?;
 
-	let plaintext = age::decrypt(&private_key, &encrypted).into_diagnostic()?;
-	tokio::fs::write(ctx.args_sub.encrypted.with_extension(""), plaintext)
+	let mut plaintext = tokio::fs::File::create_new(ctx.args_sub.encrypted.with_extension(""))
 		.await
 		.into_diagnostic()
-		.wrap_err("writing the decrypted data")?;
+		.wrap_err("opening the decrypted output")?;
 
+	let mut decrypting_reader = Decryptor::new_async(encrypted.compat())
+		.await
+		.into_diagnostic()?
+		.decrypt_async(iter::once(&private_key as _))
+		.into_diagnostic()?
+		.compat();
+
+	tokio::io::copy(&mut decrypting_reader, &mut plaintext)
+		.await
+		.into_diagnostic()
+		.wrap_err("decrypting data in stream")?;
```

Comment:
It would be useful to figure out a way to have a progress bar or status (e.g. a counter of the number of bytes processed and a percentage) for the encryption (gated on `is_terminal()`). UX-wise this is not the Unix way but we've found that people expect some kind of output that shows them that the process hasn't stalled.

Reply:
Done. `indicatif` is amazing.